### PR TITLE
Chore/funcs

### PR DESCRIPTION
**Pull request by thoward27**

### CODE REVIEW

**Review by ndaniels — APPROVED** [review]

**Comment by nishaq503** [comment]
Can we wait on this? Hamming broke for me.

File: src/categorical.rs [hunk 1]
```diff
@@ -1,6 +1,14 @@
 use rayon::prelude::*;
+use std::iter::Sum;
 
-pub fn hamming<T: PartialEq + Sync + Send>(x: &[T], y: &[T]) -> u64 {
+pub trait Categorical: PartialEq + Sync + Send + Sum {}
```
Comment (nishaq503) [inline src/categorical.rs]: Categorical does not need Sum.
This didn’t pass a cargo check for me.

**Review by nishaq503 — CHANGES_REQUESTED** [review]
Need another look at Hamming

**Review by nishaq503 — APPROVED** [review]
looks good now

**Pull request merged**